docs/examples/nodejs-read-volatile-layer.md

```diff
@@ -0,0 +1,176 @@
+
+This example shows how to retrieve partition metadata and partition data using the OLP SDK for TypeScript.
+
+## Build and run the app on Node.js
+
+Requirements:
+
+Node.js 10+
+TypeScript 3.5+
+
+Create a simple Node.js app:
+
+```
+mkdir example-app && cd example-app && npm init
+```
+
+Initialize a TypeScript project:
+
+```
+tsc --init
+```
+
+Install node types:
+
+```
+npm install --save-dev @types/node
+```
+
+Install SDK modules:
+
+```
+npm install --save @here/olp-edge-sdk-authentification @here/olp-edge-sdk-dataservice-read @here/olp-edge-sdk-dataservice-api
+```
+
+Now, everything is set to create the app.
+
+Create the index.ts file and app skeleton:
+
+```typescript
+/**
+ * Example Node.js app for reading a volatile layer from the datastore
+ */
+
+class App {
+    run() {
+        console.log("App works!");
+    }
+}
+
+const app = new App();
+app.run();
+```
+Compile and run the app:
+
+```
+tsc && node .
+```
+
+After a successful run, the console displays the following message:
+
+```
+App works!
+```
+
+## Log in to the data store
+
+The first thing you need is https://account.here.com account.
+Log in to your account and create a test app to get the app access key id and the secret access key at https://platform.here.com/admin/apps
+
+Now, you can obtain the token for requests to the datastore by using @here/olp-edge-sdk-authentification.
+
+Log in to the API. Modify our app to the following state:
+
+```typescript
+/**
+ * Example Node.js app for reading a volatile layer from the datastore
+ */
+import { UserAuth } from "@here/olp-edge-sdk-authentification";
+
+const userAuth = new UserAuth({
+    env: "here",
+    credentials: {
+        accessKeyId: "your-access-key-id",
+        accessKeySecret: "your-access-key-secret"
+    }
+});
+
+```
+
+## Context
+
+Now, you can initialize the DataStore client for `here` environment and, for example, list catalogs.
+To do so, you need `DatastoreContext`. It contains the Lookup API functionality to provide the base URLs for the services.
+
+```typescript
+/**
+ * Example Node.js app for reading a volatile layer from the datastore
+ */
+import { UserAuth } from "@here/olp-edge-sdk-authentification";
+import { DataStoreContext } from "@here/olp-sdk-dataservice-read";
+
+const userAuth = new UserAuth({
+    env: "here",
+    credentials: {
+        accessKeyId: "your-access-key-id",
+        accessKeySecret: "your-access-key-secret"
+    }
+});
+
+const context = new DataStoreContext({
+    environment: "here",
+    getToken: () => userAuth.getToken()
+});
+
+```
+
+## VolatileLayerClient
+
+When you have context, you can get catalog clients for different catalogs and read the information.
+For more information about volatile layers, see [Publish to a Volatile Layer] (https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/publishing-data-volatile.html).
+To create VolatileLayerClient, run:
+
+
+```typescript
```

Comment:
I would not put so long code blocks inside the md files, as you need to maintain and if API diverges you don't have a compile error. Only small parts explaining key concepts.
E.g. https://github.com/heremaps/here-olp-sdk-cpp/blob/master/docs/dataservice-read-catalog-example.md#retrieve-partitions-metadata